### PR TITLE
Add neo Exists request

### PR DESCRIPTION
**Pull request by gavin-norman-sociomantic**
Blocked on #64.

### CODE REVIEW

**Comment by gavin-norman-sociomantic** [comment]
Unblocked.

File: src/dhtproto/client/mixins/NeoSupport.d [hunk 1]
```diff
@@ -901,6 +937,145 @@ template NeoSupport ( )
...
+            assert(finished);
+        }
+
+        /***********************************************************************
+
+            Struct returned after an Exists request has finished.
+
+        ***********************************************************************/
+
+        private static struct ExistsResult
```
Comment (david-eckardt-sociomantic) [inline src/dhtproto/client/mixins/NeoSupport.d:988]: This could be an `enum ExistsResult { Error, Exists, NoRecord }`.

Comment (gavin-norman-sociomantic) [reply src/dhtproto/client/mixins/NeoSupport.d:988]: Yeah, I made it a struct for consistency with the other task-blocking methods.

**Comment by gavin-norman-sociomantic** [comment]
Updated: added flush calls :)

**Comment by gavin-norman-sociomantic** [comment]
Updated to use new SupportedCode from swarm v4.5.0.

File: src/dhtproto/client/request/Exists.d [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+*******************************************************************************/
+
+module dhtproto.client.request.Exists;
+
+/*******************************************************************************
+
+    Imports
+
+*******************************************************************************/
```
Comment (nemanja-boric-sociomantic) [inline src/dhtproto/client/request/Exists.d]: Suggestion: kill `Imports` with fire.

Comment (gavin-norman-sociomantic) [reply src/dhtproto/client/request/Exists.d]: Oh shit. I am ashamed!

Comment (gavin-norman-sociomantic) [reply src/dhtproto/client/request/Exists.d]: Updated.

**Review by nemanja-boric-sociomantic — APPROVED** [review]

**Pull request merged**